src/browser/fetch/request.rs

```diff
@@ -0,0 +1,282 @@
+    pub fn header(mut self, header: Header<'a>) -> Self {
+        self.headers.set(header);
+        self
+    }
+
+    /// Set HTTP method. Default method is `GET`.
+    pub const fn method(mut self, method: Method) -> Self {
+        self.method = method;
+        self
+    }
+
+    /// Set request body to provided `JsValue`. Consider using `json` or `text` methods instead.
+    ///
+    /// Note that a request using the GET or HEAD method cannot have a body.
+    pub fn body(mut self, body: JsValue) -> Self {
+        self.body = Some(body);
+
+        #[cfg(debug_assertions)]
+        match self.method {
+            Method::Get | Method::Head => {
+                error!("GET and HEAD requests shoudn't have a body");
+            }
+            _ => {}
+        }
+
+        self
+    }
+
+    /// Set request body by JSON encoding provided data.
+    /// It will also set `Content-Type` header to `application/json; charset=utf-8`.
+    ///
+    /// # Errors
+    ///
+    /// This method can fail if JSON serialization fail. It will then
+    /// return `FetchError::SerdeError`.
+    pub fn json<T: Serialize + ?Sized>(mut self, data: &T) -> Result<Self> {
+        self.headers
+            .set(header::content_type("application/json; charset=utf-8"));
+        let body = serde_json::to_string(data).map_err(FetchError::SerdeError)?;
+        self.body = Some(body.into());
+        Ok(self)
+    }
+
+    /// Set request body to a provided string.
+    /// It will also set `Content-Type` header to `text/plain; charset=utf-8`.
+    pub fn text(mut self, text: impl AsRef<str>) -> Self {
+        self.headers
+            .set(header::content_type("text/plain; charset=utf-8"));
+        self.body = Some(JsValue::from(text.as_ref()));
+        self
+    }
+
+    /// Set request mode.
+    /// It can either be `cors`, `no-cors`, `same-origin`, or `navigate`.
+    /// The default is `cors`.
+    pub const fn mode(mut self, mode: web_sys::RequestMode) -> Self {
+        self.mode = Some(mode);
+        self
+    }
+
+    /// Set request credentials.
+    /// It can either be `omit`, `same-origin`, or `include`.
+    /// The default is `same-origin`.
+    pub const fn credentials(mut self, credentials: web_sys::RequestCredentials) -> Self {
+        self.credentials = Some(credentials);
+        self
+    }
+
+    /// Set request cache mode.
+    pub const fn cache(mut self, cache: web_sys::RequestCache) -> Self {
+        self.cache = Some(cache);
+        self
+    }
+
+    /// Set request redirect mode.
+    /// It can either be `follow`, `error`, or `manual`.
+    /// The default is `follow`.
+    pub const fn redirect(mut self, redirect: web_sys::RequestRedirect) -> Self {
+        self.redirect = Some(redirect);
+        self
+    }
+
+    /// Set request referrer.
+    /// It can be either `referrer`, `client`, or a `URL`.
+    /// The default is `about:client`.
+    pub fn referrer(mut self, referrer: String) -> Self {
```

Comment:
Why `String`? I assume that user would use probably `&'static str`.
(The same for other methods which have `String` as input parameter type.)